src/spec/openapi/openapi.ts

```diff
@@ -194,20 +262,73 @@ export function openapiFromHandlers(handlers: Record<string, unknown>, options:
             },
         },
     }
+
+    // Collect schema information
+    for (const h of valuesOf(handlers)) {
+        const handler = h as EventHandler
+        if ('http' in handler && handler.http.path !== undefined) {
+            for (const key of ['body', 'query', 'path', 'headers'] as const) {
+                trackSchema(handler.http.schema[key]?.schema as JsonSchema)
+            }
+            for (const r of Object.values(handler.http.schema.responses).filter((r): r is NonNullable<typeof r> => r !== null)) {
+                trackSchema('schema' in r ? (r.schema as JsonSchema) : (r.body?.schema as JsonSchema))
+            }
+        }
+    }
+
+    // Set jsonSchemaDialect if schemas were found
+    // if (targetOpenapi === '3.1.0' && schemaFrequencies.size > 0) {
+    //     // biome-ignore lint/suspicious/noExplicitAny: <explanation>
+    //     ;(openapi as any).jsonSchemaDialect = Array.from(schemaFrequencies.entries()).reduce((a, b) => (b[1] > a[1] ? b : a))[0]
+    // }
```

Comment:
is this not needed? I'm still seeing the `schemaFrequencies` being defined, but never used

Reply:
Most tooling dont support this fully yet, keeping it there for when time progresses to when we can correctly specify our dialect :)